core/rawdb/schema.go

```diff
@@ -273,9 +273,10 @@ func IsLegacyTrieNode(key []byte, val []byte) bool {
 	return bytes.Equal(key, crypto.Keccak256(val))
 }
 
-// IsAccountTrieNode reports whether a provided database entry is an account
-// trie node in path-based state scheme.
-func IsAccountTrieNode(key []byte) (bool, []byte) {
+// ResolveAccountTrieNodeKey reports whether a provided database entry is an
+// account trie node in path-based state scheme, and returns the resolved
+// node path if so.
+func ResolveAccountTrieNodeKey(key []byte) (bool, []byte) {
```

Comment:
This would clash with hash mode trie node keys, wouldn't it? Since the trieNodeAccountPrefix might be teh same as teh first byte of the hash?

Reply:
Yes, theoretically it's possible. But in practice, we disallow to switch state scheme if the state data is already present in database. And even, if it happens, it's OK to miscount the size, it's just inspection.

Btw, it's a known issue for picking this key encoder, but compare with the benefits of this encode scheme, it's still the best.